app/experimenter/experiments/models/nimbus.py

```diff
@@ -161,6 +144,10 @@ class Filters:
     def __str__(self):
         return self.name
 
+    def set_lifecycle_state(self, lifecycle_state):
+        for name, value in lifecycle_state.value.items():
+            setattr(self, name, value)
```

Comment:
I could rename to apply. I didn't want to save here in case other changes needed to happen along with setting lifecycle state. But could probably save & generate a changelog record here to DRY up code elsewhere

Reply:
I just renamed to "apply", we can tweak to more fully saving if it's handy